x/tokenomics/keeper/token_logic_modules.go

```diff
@@ -274,15 +274,27 @@ func (k Keeper) ProcessTokenLogicModules(
 	// Execute all the token logic modules processors
 	for tlm, tlmProcessor := range tokenLogicModuleProcessorMap {
 		logger.Info(fmt.Sprintf("Starting TLM processing: %q", tlm))
-		if err := tlmProcessor(k, ctx, &service, claim.GetSessionHeader(), &application, &supplier, actualSettlementCoin, &relayMiningDifficulty); err != nil {
+		if err = tlmProcessor(k, ctx, &service, claim.GetSessionHeader(), &application, &supplier, actualSettlementCoin, &relayMiningDifficulty); err != nil {
 			return tokenomicstypes.ErrTokenomicsTLMError.Wrapf("TLM %q: %v", tlm, err)
 		}
 		logger.Info(fmt.Sprintf("Finished TLM processing: %q", tlm))
 	}
 
-	// State mutation: update the application's on-chain record
-	k.applicationKeeper.SetApplication(ctx, application)
-	logger.Info(fmt.Sprintf("updated on-chain application record with address %q", application.Address))
+	// TODO_CONSIDERATION: If we support multiple native tokens, we will need to
+	// start checking the denom here.
+	if application.Stake.Amount.LT(apptypes.DefaultMinStake.Amount) {
+		// Unbond the application because it has less than the minimum stake.
+		if err = k.applicationKeeper.UnbondApplication(ctx, &application); err != nil {
```

Comment:
Could you please add a comment that unbonding is only viable when an application is only staked for a single service.

If it is staked for multiple one, this would prevent other settlements to burn tokens.

Reply:
1. Set application `UnbondingSessionEndHeight = -1` and save here.
2. In `SettlePendingClaims()`, after `ProcessTokenLogicModules()` loop, unstake apps from 1 by lookup via claims.
3. Refactor TLM methods as **static functions** with no keeper references (probably out of scope; will open an issue).

Reply:
https://www.notion.so/buildwithgrove/Claim-Settlement-Safety-11ca36edfff6809aa951fe70a65443da

Reply:
NOTE: `UnbondingSessionEndHeight` is a `uint64` so I went with `1` instead of `-1` to keep the change small. I think this should be fine as 1 is not really a practical value for an actual unbonding session end height.